honeycomb-kernels/src/grisubal/kernel.rs

```diff
@@ -24,8 +58,746 @@ use honeycomb_core::{CMap2, CoordsFloat};
 /// ## Generics
 ///
 /// - `T: CoordsFloat` -- Floating point type used for coordinate representation.
+#[allow(
+    clippy::too_many_lines,
+    clippy::cast_possible_truncation,
+    clippy::cast_possible_wrap,
+    clippy::cast_sign_loss
+)]
 pub fn build_mesh<T: CoordsFloat>(geometry: &Geometry2<T>, grid_cell_sizes: (T, T)) -> CMap2<T> {
-    todo!()
+    // build the overlapping grid we'll modify
+    let bbox = geometry.bbox();
+    let (cx, cy) = grid_cell_sizes; // will need later
+    let (nx, ny) = (
+        (bbox.max_x / cx).ceil().to_usize().unwrap(),
+        (bbox.max_y / cy).ceil().to_usize().unwrap(),
```

Comment:
`min_x` and `min_y` are 0?

Reply:
They currently are hardcoded to 0, yes. This should not create errors since we require the geometry to be defined in positive Xs/Ys domain, but this could be easily improved with the rewrite I've done in the latest commits.